runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/quarkus/CassandraClientProducer.java

```diff
@@ -162,13 +161,20 @@ private void configureProtocolCompression(
   }
 
   private void configureMetricsSettings(ProgrammaticDriverConfigLoaderBuilder configLoaderBuilder) {
-    configLoaderBuilder.withStringList(
-        DefaultDriverOption.METRICS_NODE_ENABLED, metricsConfig.metricsNodeEnabled);
-    configLoaderBuilder.withStringList(
-        DefaultDriverOption.METRICS_SESSION_ENABLED, metricsConfig.metricsSessionEnabled);
+    // Only configure metrics to use MicroProfileMetricsFactory if a metricRegistry was provided
+    if (metricRegistry != null) {
+      configLoaderBuilder.withClass(
+          DefaultDriverOption.METRICS_FACTORY_CLASS, MicroProfileMetricsFactory.class);
+      configLoaderBuilder.withStringList(
+          DefaultDriverOption.METRICS_NODE_ENABLED, enabledNodeMetrics);
+      configLoaderBuilder.withStringList(
+          DefaultDriverOption.METRICS_SESSION_ENABLED, enabledSessionMetrics);
+    }
   }
 
   private void configureRuntimeSettings(ProgrammaticDriverConfigLoaderBuilder configLoaderBuilder) {
+    // FIXME the name is required for metrics to have the expected prefix
+    configLoaderBuilder.withString(DefaultDriverOption.SESSION_NAME, "cassandra");
```

Comment:
I'm uncomfortable about this. I think we need to investigate a way to augment metric names with an arbitrary prefix in the driver.